app/src/main/java/com/nutomic/syncthingandroid/util/Util.java

```diff
@@ -172,6 +171,38 @@ public static boolean nativeBinaryCanWriteToPath(Context context, String absolut
         return true;
     }
 
+    /**
+     * Returns if the syncthing binary would be able to write a file into
+     * the given folder given the configured access level.
+     *
+     * This uses the Android-native File API
+     */
+    public static Boolean nativeBinaryCanWriteToPath2(File externalStorageDir, String absoluteFolderPath) {
+        final String TOUCH_FILE_NAME = ".stwritetest";
+
+        // normalize path replacing ~ with external storage directory
+        // this is consistent with what $HOME is set to in SyncthingRunnable
+        final String normalizedPath = absoluteFolderPath.replaceFirst("^~", externalStorageDir.getAbsolutePath());
+
+        // Write permission test file
+        File touchFile = new File(normalizedPath, TOUCH_FILE_NAME);
+        final String touchFilePath = touchFile.getAbsolutePath();
+        try{
```

Comment:
```suggestion
        try {
```

Reply:
Please double-check whether the code follows the same convention as the current one (spacing, new lines, etc.).

Reply:
@tomasz1986 thank you for the review. I addressed the formatting issues you highlighted.